cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/KafkaClusterOperations.java

```diff
@@ -102,7 +122,30 @@ public void create(String namespace, String name, Handler<AsyncResult<Void>> han
 
             result.add(statefulSetOperations.create(kafka.generateStatefulSet(isOpenShift)));
 
-            return CompositeFuture.join(result);
+            CompositeFuture
+                .join(result)
+                .compose(res -> statefulSetOperations.waitUntilReady(namespace, kafka.getName(), 1_000, 60_000))
+                .compose(res -> {
+                    List<Future> waitPodResult = new ArrayList<>(kafka.getReplicas());
```

Comment:
Rather than joining on a bunch of `Future`s here you *could* abstract `waitUntilReady()` into a utility method that took a ready `Function` as parameter. In the current case you could all that with a method reference to `this::isReady()`. Here you could instead pass a lambda that returns the `isReady()` results for each of the replicas `&&`-ed together. I'm not saying you *should* do this, but it would mean only one timer, and fewer Futures flying around, so could well be more efficient.

Reply:
I'm fine with that but take it for another PR.